File: benchmarks/tasks/fib/fib.hs
-- fib n, the naive way: two calls and an addition per node, nothing else.
--
-- `Int` rather than the defaulted `Integer`, which is a different benchmark.

fib :: Int -> Int
fib n = if n < 2 then n else fib (n - 1) + fib (n - 2)

main :: IO ()
main = print (fib 32)

File: benchmarks/tasks/fib/fib.go
// fib n, the naive way: two calls and an addition per node, nothing else.

package main

import "fmt"

func fib(n int64) int64 {
	if n < 2 {
		return n
	}
	return fib(n-1) + fib(n-2)
}

func main() {
	fmt.Println(fib(32))
}
